### PR TITLE
Add support for multiple CIDR in vcn and fix dns label bug

### LINKED ISSUE #368: Support creating VCN with a list of CIDRs
**What would you like to be added**:
We currently create our workload cluster using the VCN created with two CIDR blocks, one for node subnet and one for pod subnet. We want to configure the VCN with a list of CIDR blocks using CAPOCI.

**Why is this needed**:
Accept a list of CIDR blocks for VCN configuration.

### PR DESCRIPTION
**Pull request by shyamradhakrishnan**
**What this PR does / why we need it**:

Add support for multiple CIDR in vcn and fix dns label bug

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #368

### CODE REVIEW

**Comment by shyamradhakrishnan** [comment]
e2e
```
[SynchronizedAfterSuite] PASSED [1.298 seconds]
[SynchronizedAfterSuite]
/home/ubuntu/go/src/github.com/oracle/cluster-api-provider-oci/test/e2e/e2e_suite_test.go:261

  Timeline >>
  STEP: Tearing down the management cluster @ 07/23/24 12:00:50.788
  << Timeline
------------------------------

Ran 7 of 28 Specs in 2678.214 seconds
SUCCESS! -- 7 Passed | 0 Failed | 0 Pending | 21 Skipped
```

**Comment by shyamradhakrishnan** [comment]
unit test
```
GOBIN=/home/ubuntu/go/src/github.com/oracle/cluster-api-provider-oci/hack/tools/bin ./scripts/go_install.sh sigs.k8s.io/controller-runtime/tools/setup-envtest setup-envtest v0.0.0-20230131074648-f5014c077fc3
kube-builder assets: /home/ubuntu/.local/share/kubebuilder-envtest/k8s/1.24.2-linux-amd64
KUBEBUILDER_ASSETS="/home/ubuntu/.local/share/kubebuilder-envtest/k8s/1.24.2-linux-amd64" go test -coverprofile=coverage.out ./...
?   	github.com/oracle/cluster-api-provider-oci	[no test files]
?   	github.com/oracle/cluster-api-provider-oci/cloud/config	[no test files]
?   	github.com/oracle/cluster-api-provider-oci/cloud/metrics	[no test files]
ok  	github.com/oracle/cluster-api-provider-oci/api/v1beta1	53.446s	coverage: 23.0% of statements
ok  	github.com/oracle/cluster-api-provider-oci/api/v1beta2	0.038s	coverage: 14.9% of statements
ok  	github.com/oracle/cluster-api-provider-oci/cloud/ociutil	0.005s	coverage: 19.6% of statements
?   	github.com/oracle/cluster-api-provider-oci/cloud/scope/mocks	[no test files]
?   	github.com/oracle/cluster-api-provider-oci/cloud/services/base	[no test files]
?   	github.com/oracle/cluster-api-provider-oci/cloud/services/base/mock_base	[no test files]
?   	github.com/oracle/cluster-api-provider-oci/cloud/services/compute	[no test files]
?   	github.com/oracle/cluster-api-provider-oci/cloud/services/compute/mock_compute	[no test files]
?   	github.com/oracle/cluster-api-provider-oci/cloud/services/computemanagement	[no test files]
?   	github.com/oracle/cluster-api-provider-oci/cloud/services/computemanagement/mock_computemanagement	[no test files]
?   	github.com/oracle/cluster-api-provider-oci/cloud/services/containerengine	[no test files]
?   	github.com/oracle/cluster-api-provider-oci/cloud/services/containerengine/mock_containerengine	[no test files]
?   	github.com/oracle/cluster-api-provider-oci/cloud/services/identity	[no test files]
?   	github.com/oracle/cluster-api-provider-oci/cloud/services/identity/mock_identity	[no test files]
?   	github.com/oracle/cluster-api-provider-oci/cloud/services/loadbalancer	[no test files]
?   	github.com/oracle/cluster-api-provider-oci/cloud/services/loadbalancer/mock_lb	[no test files]
?   	github.com/oracle/cluster-api-provider-oci/cloud/services/networkloadbalancer	[no test files]
?   	github.com/oracle/cluster-api-provider-oci/cloud/services/networkloadbalancer/mock_nlb	[no test files]
?   	github.com/oracle/cluster-api-provider-oci/cloud/services/vcn	[no test files]
?   	github.com/oracle/cluster-api-provider-oci/cloud/services/vcn/mock_vcn	[no test files]
?   	github.com/oracle/cluster-api-provider-oci/feature	[no test files]
?   	github.com/oracle/cluster-api-provider-oci/version	[no test files]
ok  	github.com/oracle/cluster-api-provider-oci/cloud/scope	230.172s	coverage: 76.1% of statements
ok  	github.com/oracle/cluster-api-provider-oci/cloud/util	0.078s	coverage: 61.1% of statements
ok  	github.com/oracle/cluster-api-provider-oci/controllers	26.750s	coverage: 58.7% of statements
ok  	github.com/oracle/cluster-api-provider-oci/exp/api/v1beta1	13.350s	coverage: 15.3% of statements
ok  	github.com/oracle/cluster-api-provider-oci/exp/api/v1beta2	0.015s	coverage: 4.8% of statements
ok  	github.com/oracle/cluster-api-provider-oci/exp/controllers	0.629s	coverage: 56.3% of statements
go tool cover -func=coverage.out -o coverage.txt
go tool cover -html=coverage.out -o coverage.html
$
```

File: cloud/scope/vcn_reconciler.go [hunk 1]
```diff
@@ -62,19 +62,13 @@ func (s *ClusterScope) GetVcnName() string {
...
 }
 
-func (s *ClusterScope) GetVcnCidr() string {
-	if s.OCIClusterAccessor.GetNetworkSpec().Vcn.CIDR != "" {
-		return s.OCIClusterAccessor.GetNetworkSpec().Vcn.CIDR
+func (s *ClusterScope) GetVcnCidrs() []string {
+	if s.OCIClusterAccessor.GetNetworkSpec().Vcn.CIDRS != nil && len(s.OCIClusterAccessor.GetNetworkSpec().Vcn.CIDRS) > 0 {
+		return s.OCIClusterAccessor.GetNetworkSpec().Vcn.CIDRS
+	} else if s.OCIClusterAccessor.GetNetworkSpec().Vcn.CIDR != "" {
+		return []string{s.OCIClusterAccessor.GetNetworkSpec().Vcn.CIDR}
```
Comment (joekr) [inline cloud/scope/vcn_reconciler.go:69]: NIT: since CIDRS supercedes CIDR thought on calling this out in variable description in types.go?

**Review by joekr — APPROVED** [review]
I missed the DNS bug. Where was that?

File: cloud/scope/vcn_reconciler.go (hunk 1 above)
Comment (shyamradhakrishnan) [reply cloud/scope/vcn_reconciler.go:69]: we have called this out in types.go? like 	// Deprecated, please use NetworkDetails.cidrs
?

Comment (shyamradhakrishnan) [reply cloud/scope/vcn_reconciler.go:69]: regarding DNS bug, please see the diff where we were creating a new spec, but ignoring setting the dns label

File: cloud/scope/vcn_reconciler.go [hunk 2]
```diff
@@ -62,19 +62,13 @@ func (s *ClusterScope) GetVcnName() string {
...
 	}
-	return VcnDefaultCidr
-}
-
-func (s *ClusterScope) VCNSpec() infrastructurev1beta2.VCN {
-	vcnSpec := infrastructurev1beta2.VCN{
-		Name: s.GetVcnName(),
-		CIDR: s.GetVcnCidr(),
-	}
```
Comment (shyamradhakrishnan) [inline cloud/scope/vcn_reconciler.go:76]: @joekr this was the bug

**Pull request merged**